Core: Remove CJS bundles, only ship ESM

ndelangen opened the pull request:
## What I did

This is the target branch of all ESM-only related work for Storybook v10

### 🦋 Canary release

<!-- CANARY_RELEASE_SECTION -->
This pull request has been released as version `0.0.0-pr-31819-sha-9ee3107e`. Try it out in a new sandbox by running `npx storybook@0.0.0-pr-31819-sha-9ee3107e sandbox` or in an existing project with `npx storybook@0.0.0-pr-31819-sha-9ee3107e upgrade`.
<details>
<summary>More information</summary>

| | |
| --- | --- |
| **Published version** | [`0.0.0-pr-31819-sha-9ee3107e`](https://npmjs.com/package/storybook/v/0.0.0-pr-31819-sha-9ee3107e) |
| **Triggered by** | @ndelangen |
| **Repository** | [storybookjs/storybook](https://github.com/storybookjs/storybook) |
| **Branch** | [`sb10/esm-only`](https://github.com/storybookjs/storybook/tree/sb10/esm-only) |
| **Commit** | [`9ee3107e`](https://github.com/storybookjs/storybook/commit/9ee3107e17e8bcadaec6e3df4ef717f6e1ffda94) |
| **Datetime** | Wed Aug 13 10:20:37 UTC 2025 (`1755080437`) |
| **Workflow run** | [16934397317](https://github.com/storybookjs/storybook/actions/runs/16934397317) |

To request a new release of this pull request, mention the `@storybookjs/core` team.

_core team members can create a new canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=31819`_
</details>
<!-- CANARY_RELEASE_SECTION -->

<!-- BENCHMARK_SECTION -->
<!-- BENCHMARK_SECTION -->


<!-- greptile_comment -->

## Greptile Summary

Target branch for transitioning Storybook v10 to ESM-only modules, marking a significant architectural shift in package format and module system.

- Initiates ESM-only architecture for Storybook v10, removing CommonJS support
- Contains preparation work for ESM transition across all Storybook packages
- Targets next branch as base for v10 development
- No canary release yet available, can be requested via `@storybookjs/core` team mention



<!-- /greptile_comment -->

JReinhold commented on code/core/bin/loader.mjs:
```diff
@@ -0,0 +1,33 @@
+    const rawSource = readFileSync(fileURLToPath(url), 'utf-8');
+    const transformedSource = transformSync(rawSource.toString(), {
+      loader: 'ts',
+      target: 'es2020',
```
I think we should instead import the `NODE_TARGET` variable we're also using for our own bundling.

JReinhold commented on code/core/package.json:
```diff
@@ -32,6 +32,9 @@
     }
   },
   "exports": {
+    "./bin/loader.mjs": {
+      "default": "./bin/loader.mjs"
+    },
```
I managed to do that here and it turned out fine. I even managed to make it all async without any problems.

https://github.com/storybookjs/storybook/commit/dfa65480cbe9a49768ca7fc485d130b7c801af3a

mrginglymus commented on scripts/build/utils/generate-types.ts:
```diff
@@ -0,0 +1,95 @@
+      return limited(async () => {
+        let timer: ReturnType<typeof setTimeout> | undefined;
+        const dtsProcess = spawn(
+          join(import.meta.dirname, '..', '..', 'node_modules', '.bin', 'jiti'),
+          [join(import.meta.dirname, 'dts-process.ts'), entryPoint],
```
This appears not to work on windows. 

```suggestion
          'node',
           [
             join(import.meta.dirname, '..', '..', 'node_modules', 'jiti', 'lib', 'jiti-cli.mjs'),
             join(import.meta.dirname, 'dts-process.ts'),
             entryPoint
           ],
```

would be a bit more portable?

JReinhold replied:
I fixed it in a slightly different way here: https://github.com/storybookjs/storybook/pull/32116

thank you ❤️

valentinpalkovic commented on scripts/build/entries/index.ts:
```diff
@@ -0,0 +1,257 @@
+        'generate-source-files.ts'
+      );
+      return new Promise((resolve, reject) => {
+        const child = exec(`jiti ${CORE_PREBUILD_SCRIPT_PATH}`, {
```
Aren't we using `execa` consistently anymore, but we are going back to native `exec` again? We should keep in mind that this may make development on Windows machines more challenging again. As long as we don't have a strong reason to ditch execa, we should ensure cross-platform support and keep using it.

valentinpalkovic commented on scripts/build/utils/dts-process.ts:
```diff
@@ -0,0 +1,67 @@
+import { writeFile } from 'node:fs/promises';
+import { dirname, join, sep } from 'node:path';
+
+import { rollup } from 'rollup';
+import { dts } from 'rollup-plugin-dts';
```
Out of curiosity. Have you tried rolldown and `rollup-plugin-dts`'s rolldown equivalent?

JReinhold replied:
it's a stretch goal of ours to try out different bundling setups, which should be more doable now that the bundling is simpler. The paths I'm currently seeing are

1. esbuild + `rollup-plugin-dts` for types 👈 what we're migrating to now. This is how the core is built in v9.
2. tsup with default dts generation 👈 this is how the non-core packages are built in v9.
3. tsup with [experimental dts generation](https://tsup.egoist.dev/#generate-declaration-file)
4. tsdown

ndelangen replied:
We have not. We're not doing a spike for new tools as part this project.

valentinpalkovic commented on scripts/build/utils/dts-process.ts:
```diff
@@ -0,0 +1,67 @@
+        compilerOptions: {
+          esModuleInterop: true,
+          baseUrl: '.',
+          jsx: JsxEmit.React,
```
Do you know if this always true? Even for building assets related to Preact, for example?

JReinhold replied:
This is copied from the core build script, so if it was a problem I'd expect we'd have heard about it for now. It doesn't look like we're building any Preact-specific JSX where we would have wanted to change this anyway.

valentinpalkovic commented on scripts/build/entries/index.ts:
```diff
@@ -0,0 +1,257 @@
+
+import type { BuildEntriesByPackageName } from '../utils/entry-utils';
+
+export const buildEntries = {
```
Could you add some JSDoc comments or a readme for other developers to elaborate on what to do if I want to build a new entry and what to consider?

valentinpalkovic commented on scripts/prepare/addon-bundle.ts:
```diff
@@ -96,7 +96,7 @@ const run = async ({ cwd, flags }: { cwd: string; flags: string[] }) => {
   if (reset) {
     await fs.emptyDir(OUT_DIR);
-    await fs.emptyDir(metafilesDir);
+    // await fs.emptyDir(metafilesDir);
```
Was this commented out for testing purposes, or can it be removed?

ndelangen replied:
This file will get deleted anyway, but the problem with this line, is that this clears all the metafiles, not just for the one package the bundle script was called for.

valentinpalkovic commented on scripts/utils/cli-step.ts:
```diff
@@ -5,9 +5,11 @@ import type { OptionSpecifier, OptionValues } from './options';
-const cliExecutable = require.resolve('../../code/core/bin/index.cjs');
-const toolboxExecutable = require.resolve('../../code/lib/cli-storybook/bin/index.cjs');
-const createStorybookExecutable = require.resolve('../../code/lib/create-storybook/bin/index.cjs');
+const cliExecutable = require.resolve('../../code/core/dist/bin/dispatcher.js');
```
Out of curiosity. Why can't `import.meta.resolve` be used here?

JReinhold replied:
it potentially can, but we're not focusing on migrating our scripts to ESM, except for the new scripts we write in `scripts/build`.

valentinpalkovic commented on test-storybooks/portable-stories-kitchen-sink/nextjs/package.json:
```diff
@@ -6,7 +6,8 @@
     "cypress": "echo 'No cypress setup yet'",
     "dev": "next dev",
-    "jest": "node --experimental-vm-modules node_modules/jest/bin/jest.js",
+    "jest": "echo 'No jest setup yet'",
+    "jest-": "node node_modules/jest/bin/jest.js",
```
What's the reason for disabling jest runs for the `portable-stories-kitchen-sink/nextjs` project?

ndelangen replied:
portable stories in jest do not work, we knew this going in to the project.

valentinpalkovic reviewed (commented):
You really are moving mountains here. Well done! I left a couple of minor comments and questions.

JReinhold commented on scripts/build/utils/dts-process.ts:
```diff
@@ -0,0 +1,67 @@
+  }
+  const { typesExternal: external } = await getExternal(process.cwd());
+
+  const dir = dirname(entryPoint).replace('src', 'dist');
```
Oh yeah that's a good one. 👍

JReinhold committed:
revert addon-docs linking workaround

ndelangen committed:
Update generateBundle to preserve process.env.NODE_ENV for production environment

ndelangen committed:
Merge branch 'norbert/esm-builders' into norbert/esm-renderers

ndelangen committed:
bust cache

ndelangen committed:
Merge pull request #31970 from storybookjs/norbert/esm-builders

ESM: Convert builders to ESM

JReinhold committed:
Apply suggestion from @greptile-apps[bot]

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

ndelangen committed:
replace node_env in runtime entries (not just global-runtime)

ndelangen committed:
Merge branch 'norbert/esm-renderers' of https://github.com/storybookjs/storybook into norbert/esm-renderers

ndelangen committed:
Add dts property to entry points in buildEntries configuration

JReinhold committed:
replace require.resolve in addon-onboarding with json import

JReinhold committed:
return channel

ndelangen committed:
Refactor entry points in package.json to simplify configuration by removing redundant types properties

ndelangen committed:
Merge branch 'sb10/esm-only' into norbert/esm-renderers

ndelangen committed:
Refactor entry points in package.json to streamline configuration by consolidating types properties into default entries

ndelangen committed:
Merge branch 'norbert/esm-renderers' of https://github.com/storybookjs/storybook into norbert/esm-renderers

JReinhold committed:
Merge pull request #32003 from storybookjs/jeppe/export-module-utils

Core: Export module utilities in common entry point

JReinhold committed:
simplify package.json import

ndelangen committed:
Merge branch 'sb10/esm-only' into norbert/esm-renderers

ndelangen committed:
bust cache

ndelangen committed:
Merge branch 'norbert/esm-renderers' of https://github.com/storybookjs/storybook into norbert/esm-renderers

JReinhold committed:
Merge pull request #32005 from storybookjs/jeppe/remove-require-onboarding

Onboarding: Replace `require.resolve` with json import

ndelangen committed:
convert framework html-vite to new bundling script & ESM-only

ndelangen committed:
make it ESM

ndelangen committed:
wip nextjs

ndelangen committed:
convert nextjs framework to esm only

ndelangen committed:
Merge pull request #31982 from storybookjs/norbert/esm-renderers

ESM: Convert renderers to ESM & new build script

ndelangen committed:
Merge branch 'next' into sb10/esm-only

ndelangen committed:
Merge branch 'sb10/esm-only' into norbert/esm-frameworks

ndelangen committed:
Refactor Next.js package.json to structure mock exports with types and defaults, and clean up build entries by removing unnecessary dts properties.

ndelangen committed:
Merge branch 'norbert/esm-frameworks' of https://github.com/storybookjs/storybook into norbert/esm-frameworks

ndelangen committed:
cleanup

ndelangen committed:
simplify

ndelangen committed:
fix: update import path for OriginalNextLegacyImage in next-legacy-image.tsx

ndelangen committed:
fix: update import paths for loader-utils and add missing module declaration

ndelangen committed:
fix: correct import path for loader-utils in get-font-face-declarations.ts

ndelangen committed:
chore: update gitHead references in package.json files across multiple addons and frameworks

ndelangen committed:
store failing sandbox as a zip, for debugging purposes

ndelangen committed:
refactor: streamline sandbox creation and artifact zipping in CircleCI config

ndelangen committed:
fix: update CircleCI config to use a variable for template in sandbox creation and zipping

ndelangen committed:
chore: remove sandbox zipping and artifact storage from CircleCI config

ndelangen committed:
chore: add zipping and artifact storage for failed sandboxes in CircleCI config

ndelangen committed:
chore: remove zipping and artifact storage for failed sandboxes in CircleCI config

JReinhold committed:
add back root preset/manager/preview entries to all addons and renderers that need them

ndelangen committed:
upgrade monorepo vite version

ndelangen committed:
convert preact-vite framework to esm only

ndelangen committed:
convert react-vite framework to esm only

ndelangen committed:
convert react-webpack5 framework to esm only

ndelangen committed:
convert server-webpack5 framework to esm only

ndelangen committed:
convert svelte-vite framework to esm only

ndelangen committed:
pin rollup

JReinhold committed:
fix format

JReinhold committed:
add require condition to framework validation

ndelangen committed:
convert vue3-vite framework to esm only

ndelangen committed:
convert web-components-vite framework to esm only

ndelangen committed:
add types for preset entrypoint for react-vite because it's used in other frameworks

ndelangen committed:
fix svelte

ndelangen committed:
bust cache

JReinhold committed:
Merge pull request #32035 from storybookjs/jeppe/allow-cjs-frameworks

Core: Support CJS presets in frameworks

ndelangen committed:
convert angular framework to esm only

JReinhold committed:
Merge branch 'sb10/esm-only' into jeppe/add-root-files

ndelangen committed:
Merge branch 'sb10/esm-only' into norbert/esm-frameworks

ndelangen committed:
cache bust

ndelangen committed:
convert ember framework to esm only

ndelangen committed:
convert nextjs-vite framework to esm only

ndelangen committed:
convert react-native-web-vite framework to esm only

ndelangen committed:
remove require from frameworks as much as possible

ndelangen committed:
convert sveltekit framework to esm only

ndelangen committed:
convert remaining frameworks to esm only and update package.json dependencies

ndelangen committed:
Merge branch 'next' into sb10/esm-only

ndelangen committed:
Merge branch 'sb10/esm-only' into norbert/esm-frameworks

ndelangen committed:
bust cache

ndelangen committed:
Update package.json files for multiple frameworks to use ES module syntax

JReinhold committed:
Merge pull request #32034 from storybookjs/jeppe/add-root-files

Addons and renderers: Fix not being able to resolve package when referenced with `getAbsolutePath`

ndelangen committed:
Refactor angular-cli-webpack.js to use ES module syntax and clean up imports

ndelangen committed:
Merge branch 'next' into sb10/esm-only

ndelangen committed:
Merge branch 'sb10/esm-only' into norbert/esm-frameworks

ndelangen committed:
Refactor post-install script to improve package name retrieval. Introduced a new function to fetch package names from paths, simplifying the logic for obtaining framework, builder, and renderer package names.

ndelangen committed:
Enhance post-install script to handle relative paths for package name retrieval. Added a check for absolute paths in the `getPackageNameFromPath` function to improve robustness.

ndelangen committed:
Enhance post-install script with name matching function for framework and builder package names. This improves the handling of package name checks, ensuring compatibility with both absolute and relative paths.

ndelangen committed:
Update post-install script to include URL handling in package name retrieval. The `getPackageNameFromPath` function now converts file URLs to paths, enhancing compatibility with various input formats.

ndelangen committed:
Add "type": "module" to package.json for ESM support in React renderer

ndelangen committed:
Update package.json to add debug-storybook script and change package references from portal to file for improved compatibility

ndelangen committed:
Refactor build-storybook script in package.json to remove debug options for a cleaner build process.

ndelangen committed:
Update preset exports in multiple framework files to include the '.js' extension for improved module resolution.

ndelangen committed:
convert codemod to esm only

ndelangen committed:
convert core-webpack to esm only

JReinhold committed:
build sveltekit preview entry, change mocks to entry points.

ndelangen committed:
convert csf-plugin to esm only

ndelangen committed:
convert create-react-app to esm only

ndelangen committed:
convert preset-react-webpack to esm only

ndelangen committed:
convert preset-server-webpack to esm only

ndelangen committed:
broaden version ranges and get them in sync across monorepo

ndelangen committed:
Merge branch 'norbert/esm-frameworks' into norbert/esm-final-packages

ndelangen committed:
Refactor preset files to use fileURLToPath for module resolution and update package.json exports to include types for improved compatibility.

ndelangen committed:
Update package.json exports in server-webpack preset to use root path for improved module resolution.

ndelangen committed:
Merge branch 'norbert/esm-final-packages' of https://github.com/storybookjs/storybook into norbert/esm-final-packages

ndelangen committed:
Refactor optimizeViteDeps export in SvelteKit preset to remove type annotation for improved clarity.

ndelangen committed:
Merge branch 'norbert/esm-frameworks' into norbert/esm-final-packages

ndelangen committed:
Update acorn and yaml package versions in yarn.lock for improved compatibility and consistency.

ndelangen committed:
Add CsfPluginOptions type export to csf-plugin index for enhanced type support

ndelangen committed:
Refactor Angular package.json and build entries to streamline client exports and remove unnecessary type annotations for improved clarity and consistency.

ndelangen committed:
Refactor getPackageNameFromPath function in postinstall script to use dynamic import for package.json, improving compatibility with JSON module loading.

ndelangen committed:
Update moduleIsAvailable function to use existsSync for module resolution, enhancing reliability in checking module availability.

ndelangen committed:
Fix import of vite-plugin-storybook-nextjs in Next.js Vite preset to use CommonJS version due to ESM output issues.

ndelangen committed:
revert it back to using require, as this needs to be sync, and many of these packages cause errors when imported statically and dynamic imports are async

ndelangen committed:
Refactor getPackageAliases function to return mapping directly, simplifying the alias configuration in Webpack.

ndelangen committed:
Refactor SWC loader configuration to simplify loading of JS config and improve browser support retrieval by directly importing getSupportedBrowsers.

ndelangen committed:
Refactor loadConfig assignment in resolveNextConfig function to handle default export more gracefully, improving compatibility with different module formats.

ndelangen committed:
Update package.json and build entries to simplify vite-plugin export and disable TypeScript declaration generation for improved build efficiency.

ndelangen committed:
Update SvelteKit package.json to simplify vite-plugin export and disable TypeScript declaration generation in build entries for enhanced efficiency.

ndelangen committed:
Fix SvelteKit preview.js export to include file extension for improved module resolution.

ndelangen committed:
Refactor vite-plugin export in package.json and build entries to streamline module resolution and disable TypeScript declaration generation for improved build efficiency.

ndelangen committed:
Add note for sb-original package in entry-utils to clarify its purpose for Next.js framework

ndelangen committed:
Remove rollup dependency from package.json in yarn-pnp directory to streamline package management.

ndelangen committed:
Refactor Angular CLI webpack configuration to replace module availability check with package resolution and enhance plugin handling for animations.

ndelangen committed:
Merge branch 'norbert/esm-frameworks' into norbert/esm-final-packages

ndelangen committed:
Update package.json and yarn.lock files to remove beta version constraints for React and ReactDOM across multiple frameworks and addons, ensuring compatibility with stable releases.

ndelangen committed:
Refactor codemod index.ts to use join for path resolution, enhancing cross-platform compatibility and simplifying code structure.

ndelangen committed:
Refactor isReactScriptsInstalled function to use camelCase for parameter naming, improving code consistency.

ndelangen committed:
Update isReactScriptsInstalled function to use 'with' option for JSON import, enhancing compatibility with module resolution.

JReinhold committed:
migrate eslint rules and config to esm

JReinhold committed:
update scripts to generate ESM

ndelangen committed:
delete unused bundle scripts

ndelangen committed:
Merge branch 'next' into sb10/esm-only

ndelangen committed:
Merge branch 'sb10/esm-only' into norbert/fix-create-storybook

ndelangen committed:
Merge branch 'sb10/esm-only' into norbert/esm-frameworks

ndelangen committed:
Merge branch 'norbert/esm-frameworks' into norbert/esm-final-packages

ndelangen committed:
Merge branch 'norbert/esm-final-packages' into norbert/esm-cleanup

JReinhold committed:
move eslint-plugin prebuild to new entries config

JReinhold committed:
migrate eslint plugin scripts to esm

ndelangen committed:
Merge pull request #32006 from storybookjs/norbert/esm-frameworks

ESM: Convert frameworks to ESM-only

ndelangen committed:
Merge pull request #32044 from storybookjs/norbert/esm-final-packages

ESM: Convert final packages in monorepo

ndelangen committed:
Merge pull request #32054 from storybookjs/norbert/esm-cleanup

ESM: Remove unused bundle scripts

JReinhold committed:
make eslint-plugin scripts cjs again, keep src as esm

JReinhold committed:
use tinyexec for executing prebuild scripts

ndelangen committed:
fixes

ndelangen committed:
Refactor preset.ts to use fileURLToPath for module resolution

JReinhold committed:
Merge branch 'sb10/esm-only' into jeppe/eslint-plugin-esm-only

JReinhold committed:
remove last bundling script, remove tsup

ndelangen committed:
Merge branch 'next' into sb10/esm-only

ndelangen committed:
Update input.tsx to import styles as a namespace for better compatibility with TypeScript

JReinhold committed:
split out build configurations to their respective packages

JReinhold committed:
simplify coverage reporter CJS workaround

JReinhold committed:
remove unused bundler properties from package.jsons

JReinhold committed:
"fix" type errors in build config imports

JReinhold committed:
fix build configs

ndelangen committed:
remove CJS references from docs about configuring storybook

ndelangen committed:
Add note on ESM requirements in main-config documentation

ndelangen committed:
Update main-config documentation to clarify ESM requirements and improve section headings

JReinhold committed:
generate metafiles named by package name, share metafile arg logic between bench script and bench stories, pass metafile arg names to benchmark bot explicitly

ndelangen committed:
Add ESM requirements for `.storybook/main.*` file in migration documentation

JReinhold committed:
cleanup

JReinhold committed:
comment on sb

JReinhold committed:
cleanup

JReinhold committed:
Merge pull request #32064 from storybookjs/jeppe/eslint-plugin-esm-only

ESLint plugin: Migrate to ESM

JReinhold committed:
fix type import

JReinhold committed:
Merge branch 'sb10/esm-only' of github.com:storybookjs/storybook into jeppe/esm-only-cleanup

JReinhold committed:
Merge pull request #32065 from storybookjs/jeppe/esm-only-cleanup

Build: Cleanup configurations and remove old build scripts

JReinhold committed:
Merge pull request #32066 from storybookjs/jeppe/simple-module-exports

Addon Vitest: Simplify coverage reporter CJS workaround

JReinhold committed:
Merge branch 'sb10/esm-only' into jeppe/fix-benchmarks

ndelangen committed:
Merge branch 'next' into sb10/esm-only

ndelangen committed:
Refactor: Replace require.resolve with import.meta.resolve for module path resolution

Updated multiple files to utilize import.meta.resolve instead of require.resolve for better compatibility with ES modules. This change enhances path resolution for mock templates and dependencies across various plugins and utilities.

ndelangen committed:
move static server asset

ndelangen committed:
chore: remove templates directory from package.json file inclusion

ndelangen committed:
fix: update mock import paths to use .ts extension for consistency

ndelangen committed:
cache bust

valentinpalkovic committed:
fix resolving mocks in esm environment

JReinhold committed:
hoist import from 'storybook/test'

ndelangen committed:
bust cache

ndelangen committed:
Merge branch 'next' into sb10/esm-only

ndelangen committed:
Merge branch 'sb10/esm-only' into fix-react-test

JReinhold committed:
disable minification, make CJS compat banner work when added multiple times to the same modules

JReinhold committed:
Merge branch 'sb10/esm-only' into jeppe/unminify

JReinhold committed:
whoops

JReinhold committed:
Merge branch 'jeppe/fix-benchmarks' of github.com:storybookjs/storybook into jeppe/fix-benchmarks

JReinhold committed:
Merge pull request #32068 from storybookjs/jeppe/fix-benchmarks

Benchmarks: Fix package benchmark reports

JReinhold committed:
Merge branch 'sb10/esm-only' into jeppe/unminify

JReinhold committed:
Apply suggestion from @greptile-apps[bot]

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

JReinhold committed:
Merge branch 'sb10/esm-only' into norbert/fix-create-storybook

JReinhold committed:
add comment about CJS compat banner

JReinhold committed:
fix script check

JReinhold committed:
bust nx cache

JReinhold committed:
remove more relative imports from create-storybook to core

ndelangen committed:
Refactor: Simplify mock path resolution and enhance external path checking

Updated the `resolveMock` function to utilize a new `isExternal` utility for determining external mock paths. This change improves clarity and maintainability. Additionally, modified the `WebpackMockPlugin` to handle mock resolution more robustly, ensuring both full paths and paths without extensions are accounted for in the mock map.

ndelangen committed:
Merge branch 'sb10/esm-only' of https://github.com/storybookjs/storybook into sb10/esm-only

ndelangen committed:
Refactor: Comment out storybook version tests in event log checker

Removed tests related to storybook version checks in the event log checker for clarity and potential future adjustments. This change simplifies the test structure while retaining the boot event logging functionality.

ndelangen committed:
Merge branch 'sb10/esm-only' into fix-react-test

ndelangen committed:
Merge branch 'sb10/esm-only' into norbert/esm-docs-changes

ndelangen committed:
Refactor: Comment out unused versions import in event log checker

Removed the import statement for versions in the event log checker script to streamline the code and prepare for potential future adjustments. This change enhances code clarity while maintaining existing functionality.

ndelangen committed:
Merge branch 'next' into sb10/esm-only

ndelangen committed:
Merge branch 'sb10/esm-only' into fix-react-test

ndelangen committed:
Merge branch 'sb10/esm-only' into norbert/esm-docs-changes

ndelangen committed:
Update docs/api/main-config/main-config.mdx

Co-authored-by: Kyle Gach <kyle.gach@gmail.com>

ndelangen committed:
Merge branch 'sb10/esm-only' into norbert/fix-create-storybook

JReinhold committed:
fix lint

JReinhold committed:
Merge branch 'norbert/fix-create-storybook' of github.com:storybookjs/storybook into norbert/fix-create-storybook

JReinhold committed:
cleanup imports

JReinhold committed:
revert module changes

ndelangen committed:
Update docs/_snippets/module-aliases-config.md

Co-authored-by: Kyle Gach <kyle.gach@gmail.com>

ndelangen committed:
Update docs/_snippets/storybook-addons-preset.md

Co-authored-by: Kyle Gach <kyle.gach@gmail.com>

JReinhold committed:
throw original import error when require fails

ndelangen committed:
Add documentation for local addon resolution in migration guide for Storybook 10

ndelangen committed:
Merge branch 'norbert/esm-docs-changes' of https://github.com/storybookjs/storybook into norbert/esm-docs-changes

ndelangen committed:
Merge pull request #32067 from storybookjs/norbert/esm-docs-changes

ESM: Update documentation to use valid ESM

ndelangen committed:
Merge branch 'sb10/esm-only' into fix-react-test

JReinhold committed:
Merge pull request #32104 from storybookjs/jeppe/improve-require-fallback

Core: Throw original import error when require fails

JReinhold committed:
Merge branch 'sb10/esm-only' of github.com:storybookjs/storybook into fix-react-test

JReinhold committed:
fix lint

JReinhold committed:
Merge branch 'sb10/esm-only' into jeppe/unminify

JReinhold committed:
remove unnecessary postbuild scripts

JReinhold committed:
fix running postbuild scripts, refactor generate bundle to support metafile generation in watch mode

JReinhold committed:
fix adding storybook config dir to vite config allow list

JReinhold committed:
make esbuild analyzer local, support any size of metafile

JReinhold committed:
Merge pull request #32090 from storybookjs/fix-react-test

React: Fix `await import(__STORYBOOK_MODULE_TEST__)` replacement

JReinhold committed:
Merge pull request #31947 from storybookjs/norbert/fix-create-storybook

ESM: No longer bundle core into create-storybook, due to ESM banner duplic…

JReinhold committed:
disable eslint for raw index.js analyzer file

JReinhold committed:
really fix lint this time

ndelangen committed:
Merge pull request #32092 from storybookjs/jeppe/unminify

Build: Stop minifying build output

ndelangen committed:
Merge branch 'next' into sb10/esm-only

ndelangen committed:
Merge branch 'sb10/esm-only' of https://github.com/storybookjs/storybook into sb10/esm-only

JReinhold committed:
change metafile output to match pacakge dir name instead of package name, so that it matches {projectName} in NX config and is correctly calculated as part of the nx output cache

JReinhold committed:
bust nx cache

JReinhold committed:
Merge branch 'sb10/esm-only' of github.com:storybookjs/storybook into sonda-analyzer

JReinhold committed:
add comment about metafile location and nx

JReinhold committed:
convert to fileURL before importing core bin

JReinhold committed:
fix mix of UNIX and POSIX paths when transforming them

JReinhold committed:
wrap absolute imported paths in pathToFileURL

JReinhold committed:
support spaces in paths on windows

JReinhold committed:
support spaces in windows paths

JReinhold committed:
fix type generation on windows and with spaces in path

JReinhold committed:
fix dispatching when path have space on windows

JReinhold committed:
fix package publishing in paths with spaces

JReinhold committed:
cleanup

ndelangen committed:
Merge branch 'next' into sb10/esm-only

JReinhold committed:
Merge branch 'sb10/esm-only' into jeppe/fix-windows-runtime

ndelangen committed:
small improvements

ndelangen committed:
Merge pull request #32109 from storybookjs/sonda-analyzer

Build: Support big metafiles in bundle analyzer

JReinhold committed:
Update scripts/knip.config.ts

Co-authored-by: greptile-apps[bot] <165735046+greptile-apps[bot]@users.noreply.github.com>

JReinhold committed:
Merge pull request #32116 from storybookjs/jeppe/fix-windows-runtime

Windows: Fix runtime and build-time errors

ndelangen committed:
Merge branch 'next' into sb10/esm-only

ndelangen committed:
Refactor CircleCI configuration to streamline job definitions and improve readability

ndelangen committed:
Add new error classes for global settings and CommonJS config support

ndelangen committed:
Refactor imports in various files to include isCI and optionalEnvToBoolean for improved configuration handling

ndelangen committed:
Merge branch 'next' into sb10/esm-only

ndelangen committed:
fix unit test

valentinpalkovic committed:
Refactor mocking utilities to enhance path resolution

ndelangen committed:
remove flaky story capture

ndelangen committed:
reduce size of

ndelangen committed:
Refactor imports across multiple files to use named import for 'ts-dedent' for consistency and clarity

ndelangen committed:
bust cache
